src/crypto/openssl/symmetric_key.cpp

```diff
@@ -62,7 +62,9 @@ namespace ccf::crypto
     CHECK1(EVP_EncryptInit_ex(ctx, NULL, NULL, key.data(), iv.data()));
     if (!aad.empty())
       CHECK1(EVP_EncryptUpdate(ctx, NULL, &len, aad.data(), aad.size()));
-    CHECK1(EVP_EncryptUpdate(ctx, cb.data(), &len, plain.data(), plain.size()));
+    if (!plain.empty())
+      CHECK1(
+        EVP_EncryptUpdate(ctx, cb.data(), &len, plain.data(), plain.size()));
     CHECK1(EVP_EncryptFinal_ex(ctx, cb.data() + len, &len));
```

Comment:
When `plain.empty()`, `cb.data()` may or may not be null, but doing anything between there and `cb.data() + len` is surely not going to end well.

Reply:
Zeroing out `len` before calling plain encryption, because previous `len` for aad never matters, as it's output buffer set to NULL.